### DISCUSSION TITLE
macOS: Quick Terminal not visible when pressing global keybind and full-screen app is active

### DISCUSSION THREAD

**Discussion by marvinruder** [ghostty-org/ghostty#2691] [Q&A] [answered]
Given that a native full-screen app (not Ghostty) is shown on the only screen, when pressing the global keybind for the quick terminal, then the full-screen app loses focus, but no quick terminal is shown.

If it is not possible to show the quick terminal on top of a native full-screen app, perhaps Ghostty can switch to a free Space and show it there?

**Comment by mitchellh** [comment]
Discussion until we determine a course of action (as noted in `CONTRIBUTING.md`). 

The only other terminal that has a global quick terminal feature is iTerm2. I suppose my first question would be: what does iTerm2 do in this scenario? 

I think the idea of switching to another space, showing it, and then switching back when it goes away is a fine behavior regardless of what iTerm2 does though.

**Reply by marvinruder** [reply]
Just tested iTerm2 – it switches to another space and opens the quick terminal there.

**Reply by mitchellh** [reply, marked as answer]
#2721. Thanks!
